### ISSUE TITLE
Cannot compile Vega-Lite Typescript project

### ISSUE THREAD

**Issue by domoritz** [rollup/rollup-plugin-typescript#30]
With this configuration

``` js
import typescript from 'rollup-plugin-typescript';
import * as ts from 'typescript';

export default {
  entry: './src/vl.ts',
  format: 'umd',

  plugins: [
    typescript({
      target: ts.ScriptTarget.ES6,
      typescript: require('typescript')
    })
  ]
}
```

I get this error

```
Could not load foo/vega-lite/src/node_modules/datalib/src/util.ts (imported by foo/vega-lite/src/util.ts): ENOENT: no such file or directory, open 'foo/vega-lite/src/node_modules/datalib/src/util.ts'
Error: Could not load foo/vega-lite/src/node_modules/datalib/src/util.ts (imported by foo/vega-lite/src/util.ts): ENOENT: no such file or directory, open 'foo/vega-lite/src/node_modules/datalib/src/util.ts'
    at /usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/src/Bundle.js:162:11
    at tryCatch (/usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:185:12)
    at invokeCallback (/usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:197:13)
    at publish (/usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:168:7)
    at publishRejection (/usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:118:3)
    at flush (/usr/local/lib/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/asap.js:87:5)
    at doNTCallback0 (node.js:428:9)
    at process._tickCallback (node.js:357:13)
    at Function.Module.runMain (module.js:459:11)
    at startup (node.js:136:18)
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki
```

Rollup tries to load a `.ts` file from the wrong root. Instead of `foo/vega-lite/src/node_modules/datalib/src/util.ts` rollup is supposed to load `foo/vega-lite/node_modules/datalib/src/util.js`

Is there a way to specify the root of the search for node modules as well as a way to tell rollup to load js files instead of ts files? 

My code is at https://github.com/vega/vega-lite/tree/dom/rollup

**Comment by Victorystick** [comment]
@domoritz I've not been able to reproduce this particular issue. The `typescript` and `node-resolve` plugins are able to locate the dependencies in the correct directories.

Instead I encounter the error;

```
Module /Users/oskar/github/vega-lite/node_modules/d3-time/index.js does not export default (imported by /Users/oskar/github/vega-lite/node_modules/datalib/src/time.js)
```

That issue may be resolved by rollup/rollup-plugin-commonjs#44.

Another problem is that the CommonJS plugin is not able to determine that `datalib/src/generate.js` and `datalib/src/util.js` has any named exports. You'll have to specify them yourself, like so

``` js
import nodeResolve from 'rollup-plugin-node-resolve';
import typescript from 'rollup-plugin-typescript';
import commonjs from 'rollup-plugin-commonjs';
import * as ts from 'typescript';

export default {
  entry: './src/vl.ts',
  dest: 'vl.js',
  moduleName: 'vl',
  sourceMap: true,

  format: 'umd',

  plugins: [
    typescript({
      target: ts.ScriptTarget.ES6,
      typescript: ts
    }),
    nodeResolve({
      jsnext: true,
      main: true
    }),
    commonjs({
      include: 'node_modules/**',

      namedExports: {
        'datalib/src/generate.js': [
          'range'
        ],

        'datalib/src/util.js': [
          'duplicate',
          'extend',
          'isArray',
          'isObject',
          'keys',
          'toMap',
          'truncate',
          'vals'
        ]
      }
    })
  ]
}
```

**Comment by domoritz** [comment]
Thanks a lot for looking into this issue! I updated the code in https://github.com/vega/vega-lite/commit/2309d1ff79745e34b44960d795a725238fcc3d7e. Unfortunately I don't get the same error but instead

```
Cannot read property 'fileExists' of undefined
TypeError: Cannot read property 'fileExists' of undefined
    at tryLoad (/Users/domoritz/Developer/UW/vega-lite/node_modules/typescript/lib/typescript.js:39331:44)
    at Object.forEach (/Users/domoritz/Developer/UW/vega-lite/node_modules/typescript/lib/typescript.js:945:30)
    at loadNodeModuleFromFile (/Users/domoritz/Developer/UW/vega-lite/node_modules/typescript/lib/typescript.js:39327:19)
    at Object.nodeModuleNameResolver (/Users/domoritz/Developer/UW/vega-lite/node_modules/typescript/lib/typescript.js:39302:36)
    at resolveId (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup-plugin-typescript/dist/rollup-plugin-typescript.cjs.js:121:37)
    at /Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/www/ROLLUP/rollup/src/utils/first.js:10:31
    at tryCatch (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:185:12)
    at invokeCallback (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:197:13)
    at publish (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/-internal.js:168:7)
    at flush (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/www/ROLLUP/rollup/node_modules/es6-promise/lib/es6-promise/asap.js:87:5)
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki
```

I'll try again when rollup/rollup-plugin-commonjs#44 is in.

Note that I'm not relying on this being fixed so take the time you need. I hope this is helpful to make rullup-typescript better. I renamed the issue.

**Comment by Victorystick** [comment]
@domoritz I just noticed that you're using TypeScript 1.8. I made a new release with some fixes for 1.8 that may address the `Cannot read property 'fileExists' of undefined` error. Mind giving the issue another go?

**Comment by domoritz** [comment]
@Victorystick Fantastic. That fixed the issue with `fileExists`. The next issue is the default export that is addressed in https://github.com/rollup/rollup-plugin-commonjs/pull/44.
